tools/convert_datasets/a2d2.py

```diff
@@ -0,0 +1,388 @@
+
+        # Partions string: [generic/path/to/file] [/] [filename]
+        img_filename = img_filepath.rpartition('/')[2]
+        ann_filename = ann_filepath.rpartition('/')[2]
+
+        if sample_idx <= train_idx_end:
+            split = 'train'
+        elif sample_idx <= val_idx_end:
+            split = 'val'
+        else:
+            split = 'test'
+
+        img_symlink_path = osp.join(a2d2_path, 'img_dir', split, img_filename)
+        ann_symlink_path = osp.join(a2d2_path, 'ann_dir', split, ann_filename)
+
+        # NOTE: Can only create new symlinks if no priors ones exists
+        try:
+            symlink(img_filepath, img_symlink_path)
+        except FileExistsError:
+            pass
+        try:
+            symlink(ann_filepath, ann_symlink_path)
+        except FileExistsError:
+            pass
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description='Convert A2D2 annotations to TrainIds')
+    parser.add_argument(
+        'a2d2_path', help='A2D2 segmentation data absolute path\
+                           (NOT the symbolically linked one!)')
+    parser.add_argument(
+        '-o', '--out-dir', help='Output path')
+    parser.add_argument(
+        '--convert', default=True, help='Convert label images')
+    parser.add_argument(
+        '--restruct', default=True, help='Restructure directory structure')
+    parser.add_argument(
+        '--type', default='cityscapes', help='Semantic label conversion type')
+    parser.add_argument(
+        '--val', default=0.02, type=float, help='Validation set sample ratio')
+    parser.add_argument(
+        '--test', default=0., type=float, help='Test set sample ratio')
+    parser.add_argument(
+        '--nproc', default=1, type=int, help='Number of process')
+    args = parser.parse_args()
+    return args
+
+
+def main():
+    '''Program for making Audi's A2D2 dataset compatible with mmsegmentation.
+
+    NOTE: The input argument path must be the ABSOLUTE PATH to the dataset
+          - NOT the symbolically linked one (i.e. data/a2d2)!
+
+    Segmentation label conversion:
+        The A2D2 labels are instance segmentations (i.e. car_1, car_2, ...),
+        while semantic segmentation requires categorical segmentations.
+
+        The function 'convert_TYPE_trainids()' converts all instance
+        segmentation to their corresponding categorical segmentation and saves
+        them as new label image files.
+
+        Conversion type options
+            A2D2: Generates segmentations using inherent categories.
+            Cityscapes: Generates segmentations according to the categories and
+                        indexing (i.e. 'trainIds') as in Cityscapes.
+
+    Directory restructuring:
+        A2D2 files are not arranged in the required 'train/val/test' directory
+        structure.
+
+        The function 'restructure_a2d2_directory' creates a new compatible
+        directory structure in the root directory, and fills it with symbolic
```

Comment:
`symbolic` is not supported for some systems. We may add option for creating symbolic or copying file.

Reply:
@xvjiarui Added a boolean command line argument `--symlink`, allowing file copy using `shutil.copyfile`